tests/k8st/tests/test_bgp_advert.py

```diff
@@ -352,17 +381,25 @@ def test_rr(self):
   asNumber: 64512
 EOF
 """)
+        # Verify clusterIP is advertised.
         svc_json = kubectl("get svc nginx-rr -n bgp-test -o json")
         svc_dict = json.loads(svc_json)
-        svcRoute = svc_dict['spec']['clusterIP']
-        retry_until_success(lambda: self.assertIn(svcRoute, self.get_routes()))
+        clusterIP = svc_dict['spec']['clusterIP']
+        retry_until_success(lambda: self.assertIn(clusterIP, self.get_routes()))
+
+        # Verify externalIP is advertised.
+        local_svc_externalips_route = "%s via 10.192.0.3" % localSvcExternalIP
+        retry_until_success(lambda: self.assertIn(local_svc_externalips_route, self.get_routes()))
 
     def test_mainline(self):
         """
         Runs the mainline tests for service ip advertisement
         - Create both a Local and a Cluster type NodePort service with a single replica.
           - assert only local and cluster CIDR routes are advertised.
           - assert /32 routes are used, source IP is preserved.
+          - add an external IP to both services.
```

Comment:
I'd recommend splitting this into two tests - one for clusterIP advertisement, and one for external IP advertisement.

Easy to understand breakages when the tests are small and focused.